entrypoint.sh

```diff
@@ -20,13 +21,26 @@ if (($(id -u) != 0)); then
        $REPOSILITE_ARGS
 # GH-1200: run as non-root user
 else
+
+  # GH-2288: Dockerfile Step-1 migration warning
+  # shellcheck disable=SC2012
+  existing_uid=$(ls -dln data | awk '{print $3}')
+  # shellcheck disable=SC2012
+  existing_gid=$(ls -dln data | awk '{print $4}')
+  if [ $existing_uid != "977" ]||[ $existing_gid != "977" ]; then
+    printf "\033[1;31mStarting with Reposilite 3.5.20 the standard user id will be 977, I will make those changes now.\033[0m\n" 1>&2
+    printf "\033[1;31mAfter 3.6.0 docker installations with user id of 999 will no longer work.\033[0m\n" 1>&2
+    printf "\033[1;31mFor more information see: https://github.com/dzikoysk/reposilite/issues/2288\033[0m\n" 1>&2
+    printf "\033[1;31mIF YOU DOWNGRADE PAST THIS POINT \"Hic sunt dracones\"\033[0m\n" 1>&2
+  fi
+
   # GH-1634: support custom user and group ids
-  GROUP_ID="${PGID:-999}"
+  GROUP_ID="${PGID:-977}"
   if ! grep -q "^reposilite" /etc/group;
   then
     addgroup --gid "$GROUP_ID" reposilite;
   fi
-  USER_ID="${PUID:-999}"
+  USER_ID="${PUID:-977}"
```

Comment:
I think that we should also include automatic permissions update, no? If we had an existing container with data being assigned to the user/group 999, then after running this one, it'll start using 977 and we'll run into permission issues :thinking: Group/user is not preserved in the volume, so we're pretty much creating it on each start in this case and that's why it may be an issue.

Reply:
> I think that we should also include automatic permissions update, no?

Line 49-50 already does the permissions change that was needed. 

> Group/user is not preserved in the volume

It is preserved when using the same volume when using docker on a linux system and also internally when using Docker Desktop (MacOS and Windows use virtualized linux systems for docker containers)
<img width="528" alt="image" src="https://github.com/user-attachments/assets/7ec7246a-064c-4328-a490-8b7d6aa67305">